### PR TITLE
Adding test for DALI FW iterators

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

### CODE REVIEW

**Comment by jantonguirao** [comment]
Build 688036

File: qa/L2_RN50_data_perf/test.sh [hunk 1]
```diff
@@ -15,6 +15,8 @@ test_body() {
     # fp16 NHWC
     python test_RN50_data_pipeline.py --gpus ${NUM_GPUS} -b 256 --workers 3 --prefetch 2 --fp16 --nhwc
     python test_RN50_data_pipeline.py --gpus ${NUM_GPUS} -b 16 --workers 3 --prefetch 11 --fp16 --nhwc
+    # FW iterators
+    python test_RN50_data_fw_iterators.py --gpus ${NUM_GPUS} -b 256 --workers 3 --prefetch 2
```
Comment (Kh4L) [inline qa/L2_RN50_data_perf/test.sh]: I think, we can have `--epochs 3` to test how the iterators "reset"

File: dali/test/python/test_RN50_data_fw_iterators.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+from __future__ import print_function
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+import glob
+import argparse
+import time
+import tensorflow as tf
+
+data_paths = ["/data/imagenet/train-jpeg"]
```
Comment (awolant) [inline dali/test/python/test_RN50_data_fw_iterators.py]: Maybe running on `val-jpeg` would be enough for this test. In the interest of time and CI resources.

File: dali/test/python/test_RN50_data_fw_iterators.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                    help='Number of epochs to run (default: 1)')
+args = parser.parse_args()
+
+N = args.gpus             # number of GPUs
+BATCH_SIZE = args.batch   # batch size
+LOG_INTERVAL = args.print_freq
+WORKERS = args.workers
+PREFETCH = args.prefetch
+FP16 = args.fp16
+NHWC = args.nhwc
```
Comment (awolant) [inline dali/test/python/test_RN50_data_fw_iterators.py]: Maybe above lines could be removed and `args` used everywhere. I don't think they introduce anything useful right now.

File: dali/test/python/test_RN50_data_fw_iterators.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        self.val = val
+        self.max_val = max(self.max_val, val)
+        self.sum += val * n
+        self.count += n
+        self.avg = self.sum / self.count
+
+import nvidia.dali.plugin as nvidia_plugin
+from nvidia.dali.plugin.mxnet import DALIClassificationIterator as MXNetIterator
+from nvidia.dali.plugin.pytorch import DALIClassificationIterator as PyTorchIterator
+from nvidia.dali.plugin.tf import DALIIterator as TensorFlowIterator
```
Comment (awolant) [inline dali/test/python/test_RN50_data_fw_iterators.py]: Maybe `imports` to the top of the file? They are not conditionally imported, so they could be grouped together.

File: dali/test/python/test_RN50_data_fw_iterators.py (hunk 2 above)
Comment (jantonguirao) [reply dali/test/python/test_RN50_data_fw_iterators.py]: there is an option to run specific number of iterations during L0, so I think it's fine. In L3 we would still go through the whole imagenet

File: dali/test/python/test_RN50_data_fw_iterators.py (hunk 4 above)
Comment (jantonguirao) [reply dali/test/python/test_RN50_data_fw_iterators.py]: I am not sure I understand. Can you elaborate?

Comment (awolant) [reply dali/test/python/test_RN50_data_fw_iterators.py]: What I meant is that canonical way for Python scripts would be to have all `imports` at the top of the file (like `PEP8` advises). Sometimes something might be conditionally imported (like inside an `if` statement) and then it makes sense to have `imports` somewhere in the middle of the file.

File: dali/test/python/test_RN50_data_fw_iterators.py (hunk 3 above)
Comment (jantonguirao) [reply dali/test/python/test_RN50_data_fw_iterators.py]: will do

File: dali/test/python/test_RN50_data_fw_iterators.py (hunk 4 above)
Comment (jantonguirao) [reply dali/test/python/test_RN50_data_fw_iterators.py]: will do

File: qa/L2_RN50_data_perf/test.sh (hunk 1 above)
Comment (jantonguirao) [reply qa/L2_RN50_data_perf/test.sh]: done

**Comment by jantonguirao** [comment]
Build 689205

**Comment by jantonguirao** [comment]
Build 692351

**Comment by jantonguirao** [comment]
Build 693932

File: qa/L0_python-self-test/test.sh [hunk 5]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash -e
 # used pip packages
-pip_packages="nose numpy opencv-python tensorflow-gpu torchvision"
+pip_packages="nose opencv-python tensorflow-gpu torchvision"
```
Comment (jantonguirao) [inline qa/L0_python-self-test/test.sh]:
```suggestion
pip_packages="nose opencv-python tensorflow-gpu torchvision mxnet-cu##CUDA_VERSION##"
```

**Pull request merged**